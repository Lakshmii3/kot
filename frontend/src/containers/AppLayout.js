import React, { Component } from 'react';
import {
    BrowserRouter as Router,
    Route,
    Link,
    Switch
} from 'react-router-dom';

import { Layout, Menu, Icon } from 'antd';

import GameboardLayout from './Gameboard';
import LobbyLayout from './Lobby';
import LoginLayout from './Login';
import NotFoundPage from './../components/NotFoundPage/NotFoundPage'

import "./AppLayout.css";

const { Header, Content, Footer } = Layout;

class AppLayout extends Component {

    state = {
        collapsed: false,
    };

    onCollapse = (collapsed) => {
        this.setState({ collapsed });
    }
    
    toggle = () => {
        this.setState({
            collapsed: !this.state.collapsed,
        });
    }

    render() {
        return (
            <Router>
                <div>
                    <Layout className="layout">
                        <Header>
                            <div className="logo" />
                            <Menu
                                theme="dark"
                                mode="horizontal"
                                defaultSelectedKeys={['1']}
                                style={{ lineHeight: '64px' }}>
                                <Menu.Item key="1">
                                    <Icon type="cluster" />
                                    <span>Lobby</span>
                                    <Link to="/lobby" />
                                </Menu.Item>
                                <Menu.Item key="2">
                                    <Icon type="desktop" />
                                    <span>Play ToK</span>
                                    <Link to="/game" />
                                </Menu.Item>
                                <Menu.Item key="3">
                                    <Icon type="login" />
                                    <span>Login</span>
                                    <Link to="/login" />
                                </Menu.Item>
                            </Menu>
                        </Header>
                        <Content style={{ margin: '24px 16px', padding: 24, background: '#fff', minHeight: 280 }}>
                            <Switch>
                                <Route exact path="/" component={LobbyLayout} />
                                <Route path="/game" component={GameboardLayout} />
                                <Route path="/lobby" component={LobbyLayout} />
                                <Route path="/login" component={LoginLayout} />
                                <Route component={NotFoundPage} />
                            </Switch>
                        </Content>
                        <Footer style={{ textAlign: 'center' }}>SE691 Software Studio - ToK</Footer>
                    </Layout>
                </div>
            </Router>
        );
    }
}

export default AppLayout;