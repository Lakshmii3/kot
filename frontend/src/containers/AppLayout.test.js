import React from 'react';
import { shallow } from 'enzyme';

import AppLayout from './AppLayout';

// verify lobby and play ToK
import {
    BrowserRouter as Router,
    Route,
    Link,
    Switch
} from 'react-router-dom';

import { Layout, Menu, Icon, Breadcrumb } from 'antd';

describe('Verify site layout template', () => {
    it('General layout should render correctly', () => {
        const errorPage = shallow(<AppLayout />);
        expect(errorPage).toMatchSnapshot();
    });

    it('renders lobby menu', () => {
        const wrapper = shallow(<AppLayout />);
        const lobby_menu = <Menu
            theme="dark"
            mode="horizontal"
            defaultSelectedKeys={['1']}
            style={{ lineHeight: '64px' }}>
            <Menu.Item key="1">
                <Icon type="cluster" />
                <span>Lobby</span>
                <Link to="/lobby" />
            </Menu.Item>
            <Menu.Item key="2">
                <Icon type="desktop" />
                <span>Play ToK</span>
                <Link to="/game" />
            </Menu.Item>
        </Menu>;
        // expect(wrapper.contains(welcome)).toBe(true);
        // expect(wrapper.contains(welcome)).toEqual(true);
        expect(wrapper).toContainReact(lobby_menu);
    });

    it('renders lobby menu', () => {
        const wrapper = shallow(<AppLayout />);
        const lobby_menu = <Menu
            theme="dark"
            mode="horizontal"
            defaultSelectedKeys={['1']}
            style={{ lineHeight: '64px' }}>
            <Menu.Item key="1">
                <Icon type="cluster" />
                <span>Lobby</span>
                <Link to="/lobby" />
            </Menu.Item>
            <Menu.Item key="2">
                <Icon type="desktop" />
                <span>Play ToK</span>
                <Link to="/game" />
            </Menu.Item>
        </Menu>;
        // expect(wrapper.contains(welcome)).toBe(true);
        // expect(wrapper.contains(welcome)).toEqual(true);
        expect(wrapper).toContainReact(lobby_menu);
    });
})