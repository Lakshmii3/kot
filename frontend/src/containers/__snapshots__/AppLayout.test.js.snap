// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify site layout template General layout should render correctly 1`] = `
<BrowserRouter>
  <div>
    <Adapter
      className="layout"
    >
      <Adapter>
        <div
          className="logo"
        />
        <Menu
          defaultSelectedKeys={
            Array [
              "1",
            ]
          }
          mode="horizontal"
          style={
            Object {
              "lineHeight": "64px",
            }
          }
          theme="dark"
        >
          <MenuItem
            key="1"
          >
            <Icon
              type="cluster"
            />
            <span>
              Lobby
            </span>
            <Link
              to="/lobby"
            />
          </MenuItem>
          <MenuItem
            key="2"
          >
            <Icon
              type="desktop"
            />
            <span>
              Play ToK
            </span>
            <Link
              to="/game"
            />
          </MenuItem>
        </Menu>
      </Adapter>
      <Adapter
        style={
          Object {
            "background": "#fff",
            "margin": "24px 16px",
            "minHeight": 280,
            "padding": 24,
          }
        }
      >
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/game"
          />
          <Route
            component={[Function]}
            path="/lobby"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </Adapter>
      <Adapter
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        SE691 Software Studio - ToK
      </Adapter>
    </Adapter>
  </div>
</BrowserRouter>
`;
